src/main/java/net/dv8tion/jda/api/interactions/commands/PrivilegeConfig.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.interactions.commands;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.SelfUser;
+import net.dv8tion.jda.api.interactions.commands.privileges.IntegrationPrivilege;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents the Privilege-Config for this integration in a Guild.
+ * <br>Can be used to get a List of {@link net.dv8tion.jda.api.interactions.commands.privileges.IntegrationPrivilege IntegrationPrivileges}
+ * applied on the application itself and on each command.
+ *
+ * @see Guild#retrieveCommandPrivileges()
+ */
+public class PrivilegeConfig
+{
+    private final Guild guild;
+    private final Map<String, List<IntegrationPrivilege>> privileges;
+
+    public PrivilegeConfig(@Nonnull Guild guild, @Nonnull Map<String, List<IntegrationPrivilege>> privileges)
+    {
+        this.guild = guild;
+        this.privileges = Collections.synchronizedMap(privileges);
```

Comment:
Why is this synchronized? The map shouldn't get modified after creation anyway. Maybe you wanted `unmodifiableMap` here instead.

Reply:
lmao whoops, meant `unmodifiableMap` instead yes.